drf_kit/models/availability_models.py

```diff
@@ -0,0 +1,58 @@
+import logging
+
+from django.db import models
+from django.utils.translation import gettext as _
+
+from drf_kit import managers
+from drf_kit.models.base_models import BaseModel
+
+logger = logging.getLogger(__name__)
+
+
+class AvailabilityModelMixin(models.Model):
+    starts_at = models.DateTimeField(
+        verbose_name=_("starts at"),
+        blank=True,
+        null=True,
+        default=None,
+    )
+    ends_at = models.DateTimeField(
+        verbose_name=_("ends at"),
+        blank=True,
+        null=True,
+        default=None,
+    )
+
+    class Meta:
+        abstract = True
+        indexes = [
+            models.Index(fields=["starts_at"]),
+            models.Index(fields=["ends_at"]),
```

Comment:
I'm not sure about the gain in performance by adding indexes with `datetime` fields but is a feeling that is not so effective I have no data to prove that

Reply:
I do not have much XP with that either. So we'll see.
I know that [PSQL has table partitioning](https://www.postgresql.org/docs/current/ddl-partitioning.html) that would come in handy in this case, but it looks like Django does not support it yet.